Release/include/pplx/threadpool.h

```diff
@@ -97,6 +101,19 @@ class threadpool
         return m_service;
     }
 
+    std::shared_ptr<web::http::client::details::asio_connection_pool> obtain_connection_pool(const std::string &base_uri, std::function<std::shared_ptr<web::http::client::details::asio_connection_pool>()> connection_pool_generator)
```

Comment:
The implementation for this function needs to go into `Release\src\pplx\threadpool.cpp` file instead of the header.

Since we're only using this function with a single callback, it seems simpler to me if the single callback was inlined into this function. Instead of taking the base_uri, this function could just take the `http_client_config` by reference and determine whether to make a new pool or to reuse a cached pool based on the `base_uri()` and `ssl_context_callback`.